docs/index.rst

```diff
@@ -283,6 +316,47 @@ With the above report, Betelgeuse performs the following:
     to run ``test-command`` command before ``test-run`` to make sure all
     required test cases are created/updated accordingly.
 
+The test-run command allows setting custom fields in order to better define the
+environment. There are two way to define custom fields the ``key=value`` and
+JSON formats. The former is a shortcut when you want to define plain strings as
+the value of a custom field, on the other hand, the later is better suited when
+the type of the custom field matters, for example, if a custom field expects a
+boolean as a value.
+
+Below is an example using the ``key=value`` format:
+
+.. code-block:: console
+
+    $ betelgeuse test-run \
+        --path sample_project/results/sample-junit-result.xml \
+        --test-run-id regression_test_run_1 \
+        --test-template-id Empty
+        --user testuser1 \
+        --source-code-path sample_project/tests/ \
+        --custom-fields arch=x8664 \
+        --custom-fields variant=server \
+        PROJECT_CLOUD
+
+Now an example using JSON format:
+
+.. code-block:: console
+
+    $ betelgeuse test-run \
+        --path sample_project/results/sample-junit-result.xml \
+        --test-run-id regression_test_run_1 \
+        --test-template-id Empty
+        --user testuser1 \
+        --source-code-path sample_project/tests/ \
+        --custom-fields '{"isautomated":true,"arch":"x8664"}' \
+        PROJECT_CLOUD
+
+.. warning::
+
+    Betelgeuse does not validate the value of the custom fields, you are
+    responsible for passing the expected values. If an unnexpected value is
+    found the command will fail with a stack trace showing what need to be
+    fixed.
+
```

Comment:
How about:

Make sure to pass the right value for the custom fields as Betelgeuse does not validate them.  If an unexpected value is found, the command will fail with a stack trace showing the error.